core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -266,21 +268,37 @@ class PlaintextConsumerTest extends BaseConsumerTest {
   def testSeek() {
     val consumer = this.consumers(0)
     val totalRecords = 50L
-    sendRecords(totalRecords.toInt)
+    sendRecords(totalRecords.toInt, tp)
+    sendCompressedMessages(totalRecords.toInt, tp, totalRecords.toInt)
     consumer.assign(List(tp).asJava)
 
     consumer.seekToEnd(tp)
-    assertEquals(totalRecords, consumer.position(tp))
+    assertEquals(totalRecords * 2, consumer.position(tp))
     assertFalse(consumer.poll(totalRecords).iterator().hasNext)
 
     consumer.seekToBeginning(tp)
     assertEquals(0, consumer.position(tp), 0)
     consumeAndVerifyRecords(consumer, numRecords = 1, startingOffset = 0)
 
-    val mid = totalRecords / 2
-    consumer.seek(tp, mid)
-    assertEquals(mid, consumer.position(tp))
-    consumeAndVerifyRecords(consumer, numRecords = 1, startingOffset = mid.toInt, startingKeyAndValueIndex = mid.toInt)
+    // Test seek non-compressed message
+    val quarter = totalRecords / 2
```

Comment:
may want to rename these variables. quarter and threeQuarters are slightly confusing